### PR TITLE
AndroidX migration

### PR DESCRIPTION
**Pull request by AntonPieper**
This migrates the codebase to AndroidX,  because `com.android.support:appcompat-v7` is [deprecated](https://developer.android.com/topic/libraries/support-library/packages)

I also raised the minSdkVersion to 14 so that the project compiles. I think this is reasonable, because Android 4.0 [(SDK Version 14, released 2011)](https://developer.android.com/tools/releases/platforms#4.0) is so far back in time that I don't think anyone has a phone with an older android version _(**and** wants to run ShaderEditor on that)_.

**Edit:** Maybe The compile error could have been caused by my `gradle.properties`, I had set `android.enableJetifier=true`. Either way, I think that raising the minSdkVersion is better, because the [`<uses-sdk>`](https://github.com/markusfisch/ShaderEditor/blob/d953cde95dff8bef042f86ecb88aead2b3a223e3/app/src/main/AndroidManifest.xml#L25)-approach seem kind of hacky.

### CODE REVIEW

**Comment by markusfisch** [comment]
First of all thanks a lot for breaking things up into individual PRs! 👍 

Just one tiny little thing: could you please add a `gradle.properties` files with the following content to make this commit compile?
```
android.enableJetifier=true
android.useAndroidX=true
```

Of course, I could add it in a subsequent commit. But I think it would be better if this commit would compile 😉

**Comment by AntonPieper** [comment]
Yeah, the `android.useAndroidX=true` setting is required. I want to note that this would require the file to be removed from the [`.gitignore`](https://github.com/markusfisch/ShaderEditor/blob/d953cde95dff8bef042f86ecb88aead2b3a223e3/.gitignore#L9). With the other PRs still open, I think it makes more sense to merge/reject them before this one as it's less work to convert them all at once instead of one by one (would need to merge/rebate the changes into each branch).

**Comment by markusfisch** [comment]
Yes, `gradle.properties` would have to be removed from `.gitignore` now, but you could also use the `-f` flag to force add it, if that's easier for now:
```
$ git add -f gradle.properties
```

Maybe I'm wrong, but I _think_ you can add this to this branch (and push it into here) without needing to touch the other PRs 🤔 Could you please have a try? I cannot add something to this PR, unfortunately. And please excuse me for being so picky 🙈 I really appreciate your work! I just maybe care a bit too much about the commit history, and we're _so_ close to the perfect commit 😉

File: gradle.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+android.enableJetifier=false
```
Comment (AntonPieper) [inline gradle.properties:1]: Jetifier is not needed according to Build Analyzer

**Comment by AntonPieper** [comment]
I have added the `gradle.properties` file to the repository now. You are actually correct, the other PRs are unaffected (except #149, because it extends `AppCompatEditText` [here](https://github.com/AntonPieper/ShaderEditor/blob/8a081329fee19b64692b3355cede7c64b193115c/app/src/main/java/de/markusfisch/android/shadereditor/widget/LineNumberEditText.java#L13))

**Comment by markusfisch** [comment]
Thanks a lot, and thanks again for adding the `gradle.properties` 👍

**Comment by AntonPieper** [comment]
I have just noticed that the Android code style separates the `import android.*` and `import androidx.*` imports, should this behaviour be enforced, or should I edit the `.editorconfig` file to not put blank lines in between? With the blank lines, reformatting changes basically all files.

File: gradle.properties (hunk 1 above)
Comment (markusfisch) [reply gradle.properties:1]: Hm, `false` doesn't compile for me 🤔 Somehow Gradle thinks there are still legacy support libraries:

> WARNING:Your project has set `android.useAndroidX=true`, but configuration 
> `:app:debugRuntimeClasspath` still contains legacy support libraries,
> which may cause runtime issues.
>
> This behavior will not be allowed in Android Gradle plugin 8.0.                 
>
> Please use only AndroidX dependencies or set `android.enableJetifier=true` in the
> `gradle.properties` file to migrate your project to AndroidX (see 
> https://developer.android.com/jetpack/androidx/migrate for more info)

I just made another commit that sets it to `true` to make it compile. But I should have a deeper look at this…

**Comment by markusfisch** [comment]
No, that's okay, I've got no hard feelings about how the imports are formatted as long as it's consistent.

I would just want to avoid double blank lines.

**Pull request merged**